lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+module.exports = {
+  compileRequestValidators() {
+    const apiGateway = this.serverless.service.provider.apiGateway || {};
+    const models = {};
+    // Models are defined in the provider and can be used in http events later
+    if (apiGateway.requestSchemas) {
+      for (const [schemaId, schemaConfig] of _.entries(apiGateway.requestSchemas)) {
+        let modelName;
+        let validatorName;
+        let description;
+        let definition;
+
+        // If schema is not defined this will try to map resourceDefinition as the schema
+        if (!schemaConfig.schema) {
+          definition = schemaConfig;
+        } else {
+          definition = schemaConfig.schema;
+        }
+
+        const resourceName = _.upperFirst(_.camelCase(schemaId));
+        const modelLogicalId = this.provider.naming.getModelLogicalId(
+          resourceName,
+          'any',
+          'application/json'
+        );
```

Comment:
I think we need something more solid, in current form we're using method which was strictly designed to resolve logical id for specific HTTP endpoint and content type, without any endpoint and content type context. 

I know I suggested to use same method, but that just doesn't look good.

I propose to rename current method to `getEndpointModelLogicalId`, and introduce another `getModelLogicalId` that takes just `schemaId` and resolves id as:

```javascript
return `ApiGateway${_.upperFirst(_.camelCase(schemaId))}Model`;
```